### PR TITLE
units: Add `is_satisfied_by` locktime tests

### LINKED ISSUE #4587: New Mutants Found
Displaying up to the first 10 mutants:

units/src/locktime/absolute.rs:94:9: replace Height::is_satisfied_by -> bool with true
units/src/locktime/absolute.rs:94:9: replace Height::is_satisfied_by -> bool with false
units/src/locktime/absolute.rs:94:60: replace + with - in Height::is_satisfied_by
units/src/locktime/absolute.rs:94:60: replace + with * in Height::is_satisfied_by
units/src/locktime/absolute.rs:95:34: replace <= with > in Height::is_satisfied_by
units/src/locktime/absolute.rs:241:9: replace MedianTimePast::is_satisfied_by -> bool with true
units/src/locktime/absolute.rs:241:9: replace MedianTimePast::is_satisfied_by -> bool with false
units/src/locktime/absolute.rs:241:14: replace <= with > in MedianTimePast::is_satisfied_by

For the complete list, please check the [mutants.out artifact](https://github.com/rust-bitcoin/rust-bitcoin/actions/runs/15370008908).

### PR DESCRIPTION
**Pull request by jamillambert**
Weekly mutation testing found new mutants in both `Height::is_satisfied_by` and `MedianTimePast::is_satisfied_by` functions.

Test these two functions and kill the mutants.

Closes #4587

### CODE REVIEW

File: units/src/locktime/absolute.rs [hunk 1]
```diff
@@ -607,4 +607,34 @@ mod tests {
...
+        let locktime = Height::from_u32(100).unwrap();
+        assert!(locktime.is_satisfied_by(chain_tip));
+
+        let chain_tip = Height::from_u32(100).unwrap();
+        let locktime = Height::from_u32(101).unwrap();
+        assert!(locktime.is_satisfied_by(chain_tip));
+
+        let chain_tip = Height::from_u32(100).unwrap();
+        let locktime = Height::from_u32(102).unwrap();
+        assert!(!locktime.is_satisfied_by(chain_tip));
```
Comment (tcharding) [inline units/src/locktime/absolute.rs]: `chain_tip` is the same in each assertion so I rekon only define it once. And can we have a comment on the middle test case

```
// lock is satisfied if transaction can go in the next block.
```

File: units/src/locktime/absolute.rs [hunk 2]
```diff
@@ -607,4 +607,34 @@ mod tests {
...
+    }
+
+    #[test]
+    fn median_time_past_is_satisfied_by() {
```
Comment (tcharding) [inline units/src/locktime/absolute.rs:627]: Here we could use a single `mpt` also. I'll let you write the required code comment equivalent to the other tet (but time is handled differently to height). The aim being to test whether the current code and docs are clear enough because if you can't write the code comment then we need to improve the modules more.

File: units/src/locktime/absolute.rs (hunk 1 above)
Comment (jamillambert) [reply units/src/locktime/absolute.rs]: done

File: units/src/locktime/absolute.rs (hunk 2 above)
Comment (jamillambert) [reply units/src/locktime/absolute.rs:627]: done

Comment (jamillambert) [reply units/src/locktime/absolute.rs:627]: When writing the test I did have to look into median time past to find out why is was not `locktime <= mtp +1`. And if I  unterstand correctly it is possible for the next block to have the same mtp. Opposed to the next block height, which by definition has to have a height of +1.

Comment (tcharding) [reply units/src/locktime/absolute.rs:627]: Close but not quite. Its different conceptually because height we can predict (next block is +1) but MTP of the next block is not known. Technically what happens is that when verifying a block the MTP of the _previous_ block is used, that's why we can use <=

Does that make sense?

**Review by tcharding — APPROVED** [review]
ACK 2fa5c062d5e07580bdb7ea5e4c58e4607c716ecc

**Comment by tcharding** [comment]
I acked because the mtp is testing the boundary cases which is good even if the exact values don't make heaps of sense. I think the comments are close enough, these concepts are not easy to explain in a single code comment anyways.

File: units/src/locktime/absolute.rs (hunk 2 above)
Comment (jamillambert) [reply units/src/locktime/absolute.rs:627]: The issue was not the = in the <=, it was the +1. i.e. mtp can't go down with new blocks so the <= makes sense. But if the next block has to have an mtp > then the previous block as the block height does then it should be `<= mtp +1`. So the reason the locktime is valid if equal to the `height +1` but just `mtp` not `mtp + 1` is because the next block could have the same mtp, i.e. for this case a locktime of `mtp + 1` is not valid for this block.

**Comment by apoelstra** [comment]
Nice. This sort of "boundary case testing" is exactly what we want the mutation tester to find.

**Review by apoelstra — APPROVED** [review]
ACK 2fa5c062d5e07580bdb7ea5e4c58e4607c716ecc; successfully ran local tests
